events/events_handler.go

```diff
@@ -0,0 +1,309 @@
+	}
+}
+
+// determines if an event occurs within the time set in the request object
+func checkIfEventSatisfiesTimeRequest(request *Request, e *Event) bool {
+	startTime := request.StartTime
+	endTime := request.EndTime
+	eventTime := e.TimeOfEvent
+	if !startTime.IsZero() {
+		if startTime.Sub(eventTime) > 0 {
+			return false
+		}
+	}
+	if !endTime.IsZero() {
+		if endTime.Sub(eventTime) < 0 {
+			return false
+		}
+
+	}
+	return true
+}
+
+// sorts and grabs the last N events, as specified in the request object,
+// or if there are fewer than N events, returns them all
+func (self *Events) getLastNEvents(request *Request) error {
+	n := request.LastNEvents
+	if n >= self.Len() || n <= 0 {
+		return nil
+	}
+	sort.Sort(self)
+	self.EventList = self.EventList[self.Len()-n-1:]
+	return nil
+}
+
+// called by GetEvents to gather ContainerCreation events that have already
+// occurred and populate the Events.EventList slice with them
+func (self *Events) gatherCreatedContainers(request *Request) error {
+	return nil
+}
+
+// called by GetEvents to gather ContainerDeletion events that have already
+// occurred and populate the Events.EventList slice with them
+func (self *Events) gatherDeletedContainers(request *Request) error {
+	return nil
+}
+
+// called by GetEvents to gather oomInstance events that have already
+// occurred and populate the Events.EventList slice with them
+func (self *Events) gatherOoms(request *Request) error {
+	// out is a user-provided channel from which the user can read incoming
+	// OomInstance objects
+	outStream := make(chan *oomparser.OomInstance)
+	oomLog, err := oomparser.New()
+	if err != nil {
+		glog.Infof("couldn't make a new oomparser. %v", err)
+		return err
+	} else {
+		timeout := make(chan bool, 1)
+		go func() {
+			time.Sleep(1 * time.Second)
+			timeout <- true
+		}()
+		err := oomLog.StreamOoms(outStream)
+		if err != nil {
+			glog.Errorf("%v", err)
+			return err
+		}
+		select {
+		case oomInstance := <-outStream:
+			newOomEvent := OomInstanceToEvent(oomInstance)
+			withinTimeRange := checkIfEventSatisfiesTimeRequest(request, newOomEvent)
+			if withinTimeRange {
+				self.EventList = append(self.EventList, newOomEvent)
+			}
+		case <-timeout:
+			glog.Errorf("timed outwhile gathering Ooms!")
+		}
+	}
+	return nil
+}
+
+// called by Watch to gather ContainerCreation events as they occur
+// populate the Events.EventList slice with them
+func (self *Events) monitorCreatedContainers(eventChannel chan *Event, request *Request) error {
+	return nil
+}
+
+// called by Watch to gather ContainerDeletion events as they occur
+// populate the Events.EventList slice with them
+func (self *Events) monitorDeletedContainers(eventChannel chan *Event, request *Request) error {
+	return nil
+}
+
+// called by Watch to gather OomInstanceCat events as they occur
+// populate the Events.EventList slice with them
+func (self *Events) monitorOoms(eventChannel chan *Event, request *Request) error {
```

Comment:
We shouldn't be monitoring for OOMs ourselves. Let another component do that and feed them to you with AddEvent() :)